### ISSUE TITLE
texts_from_df distilbert performance compared to bert

### ISSUE THREAD

**Issue by lukaj1999** [amaiya/ktrain#386]
Seeing very long training times and poor accuracy scores when running distilbert.  Seeing better w/ bert - and much better with nbsvm.

Running on Azure Databricks with GPU cluster -

Requirement already satisfied: tensorflow in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (2.5.0)
Requirement already satisfied: numpy~=1.19.2 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.19.2)
Requirement already satisfied: opt-einsum~=3.3.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (3.3.0)
Requirement already satisfied: keras-nightly~=2.5.0.dev in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (2.5.0.dev2021032900)
Requirement already satisfied: termcolor~=1.1.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.1.0)
Requirement already satisfied: protobuf>=3.9.2 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (3.13.0)
Requirement already satisfied: typing-extensions~=3.7.4 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (3.7.4.3)
Requirement already satisfied: wrapt~=1.12.1 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.12.1)
Requirement already satisfied: flatbuffers~=1.12.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.12)
Requirement already satisfied: gast==0.4.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (0.4.0)
Requirement already satisfied: absl-py~=0.10 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (0.11.0)
Requirement already satisfied: google-pasta~=0.2 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (0.2.0)
Requirement already satisfied: tensorboard~=2.5 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (2.5.0)
Requirement already satisfied: six~=1.15.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.15.0)
Requirement already satisfied: wheel~=0.35 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (0.35.1)
Requirement already satisfied: grpcio~=1.34.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.34.1)
Requirement already satisfied: astunparse~=1.6.3 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.6.3)
Requirement already satisfied: keras-preprocessing~=1.1.2 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (1.1.2)
Requirement already satisfied: tensorflow-estimator<2.6.0,>=2.5.0rc0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (2.5.0)
Requirement already satisfied: h5py~=3.1.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorflow) (3.1.0)
Requirement already satisfied: setuptools in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from protobuf>=3.9.2->tensorflow) (50.3.1.post20201107)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorboard~=2.5->tensorflow) (1.8.0)
Requirement already satisfied: markdown>=2.6.8 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorboard~=2.5->tensorflow) (3.3.3)
Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorboard~=2.5->tensorflow) (0.6.1)
Requirement already satisfied: google-auth<2,>=1.6.3 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorboard~=2.5->tensorflow) (1.22.1)
Requirement already satisfied: requests<3,>=2.21.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorboard~=2.5->tensorflow) (2.24.0)
Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorboard~=2.5->tensorflow) (0.4.2)
Requirement already satisfied: werkzeug>=0.11.15 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from tensorboard~=2.5->tensorflow) (1.0.1)
Requirement already satisfied: cachetools<5.0,>=2.0.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (4.2.2)
Requirement already satisfied: rsa<5,>=3.1.4; python_version >= "3.5" in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (4.7.2)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (0.2.8)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (1.25.11)
Requirement already satisfied: idna<3,>=2.5 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (2.10)
Requirement already satisfied: chardet<4,>=3.0.2 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (3.0.4)
Requirement already satisfied: certifi>=2017.4.17 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from requests<3,>=2.21.0->tensorboard~=2.5->tensorflow) (2021.5.30)
Requirement already satisfied: requests-oauthlib>=0.7.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.5->tensorflow) (1.3.0)
Requirement already satisfied: pyasn1>=0.1.3 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from rsa<5,>=3.1.4; python_version >= "3.5"->google-auth<2,>=1.6.3->tensorboard~=2.5->tensorflow) (0.4.8)
Requirement already satisfied: oauthlib>=3.0.0 in /databricks/conda/envs/databricks-ml-gpu/lib/python3.8/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.5->tensorflow) (3.1.0)

**Comment by amaiya** [comment]
I was not able to reproduce this problem.  If you are able to reproduce this on Google colab, please send me a link to the notebook, so it can be investigated.

**Comment by lukaj1999** [comment]
Closing.  Looks like the issue was approx. 30% of training data was randomly labeled.

**Comment by lukaj1999** [comment]
Bad data in training set - affecting ability of transfer learning models
